src/main/java/minicraft/level/Level.java

```diff
@@ -370,6 +371,13 @@ public void tick(boolean fullTick) {
 			}
 
 			entitiesToAdd.remove(entity);
+
+			if (getLevelName(depth).equals("Lava")) {
+				AchievementsDisplay.setAchievement("minicraft.achievement.lowest_caves",true);
+			}
+			if (getLevelName(depth).equals("Dungeon")) {
+				AchievementsDisplay.setAchievement("minicraft.achievement.obsidian_dungeon",true);
+			}
```

Comment:
This should be moved to a function called when moving from level to level.

Reply:
Nope, put this code in a method that changes your level. I think it it should be in Level.java